src/common/utils/use-store.ts

```diff
@@ -0,0 +1,58 @@
+import { PublicSubject } from '../../services/stores/common/types';
+import { useGlobalStore } from '../../services/stores/global';
+import { Store, StoreType, StoresTypes } from '../types/stores.types';
+
+const stores = {
+  [StoreType.GLOBAL]: useGlobalStore,
+};
+
+/**
+ * @function subscribe
+ * @description Subscribes to a subject and either update the value of the property each time there is a change, or call the callback that provides a custom behavior to the subscription
+ * @param name The name of the property to be updated in case there isn't a callback
+ * @param subject The subject to be subscribed
+ * @param callback The callback to be called each time there is a change
+ */
+function subscribeTo<T>(
+  name: string,
+  subject: PublicSubject<T>,
+  callback?: (value: T) => void,
+): void {
+  subject.subscribe(this, () => {
+    this[name] = subject.value;
+
+    if (callback) {
+      callback(subject.value);
+    }
+  });
+
+  this.unsubscribeFrom.push(subject.unsubscribe);
+
+  if (this.requestUpdate) this.requestUpdate();
```

Comment:
Is it used by lit components?

Reply:
Yes, it is. In both the base class of our components and the base class of our web components, there is something like `protected useStore = useStore.bind(this)`. Since it is predictable what to do when some value updates when dealing with the web components (that is, you need to update the visuals based on the new value too), I call `this.requestUpdate` here to makes this happen. With our normal components, there isn't such an obvious flow to follow, so that's why there isn't an equivalent call to it

Reply:
Fine!